### ISSUE TITLE
[JENKINS-43907] Git Creation produces error in console after successful pipeline creation

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/blueocean-plugin#4068]
<p><b>Repro Steps</b></p>

<ol>
	<li>Click "New Pipeline"</li>
	<li>Choose "Git"</li>
	<li>Enter URL:  <a href="https://bitbucket.org/cliffmeyers/jenkins-test-public-1.git" class="external-link" target="_blank" rel="nofollow noopener">https://bitbucket.org/cliffmeyers/jenkins-test-public-1.git</a></li>
	<li>Press "Create"</li>
</ol>

<p><b>Expected Results</b></p>
<ul class="alternate" type="square">
	<li>User is taken to Activity tab displaying two builds</li>
</ul>

<p><b>Actual Results</b></p>
<ul class="alternate" type="square">
	<li>User is taken to the tab, but errors in console</li>
</ul>

<p><b>Error</b></p>

<pre>
Uncaught (in promise) TypeError: Cannot read property <span class="code-quote">'self'</span> of undefined
    at ActivityService.bunkerKey (blueocean.js:5397)
    at ActivityService.setItem (blueocean.js:5440)
    at executeAction (blueocean.js:63849)
    at ActivityService.res (blueocean.js:63821)
    at blueocean.js:5684
</pre>

<p><b>Notes</b><br/>
This error seems to occur because the method receives the ObservableValue instance, instead of the actual object (ObservableValue.value) </p>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/cliffmeyers.png" title="cliffmeyers's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=cliffmeyers">cliffmeyers</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-43907" target="_blank">Git Creation produces error in console after successful pipeline creation</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/cliffmeyers.png" title="cliffmeyers's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=cliffmeyers">cliffmeyers</a>
<li><b>status</b>: Closed
<li><b>priority</b>: Minor
<li><b>component(s)</b>: blueocean-plugin
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2017-06-09T14:42:33+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 3
<li><b>imported</b>: 2025-12-01
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p><b>Repro Steps</b></p>

<ol>
	<li>Click "New Pipeline"</li>
	<li>Choose "Git"</li>
	<li>Enter URL:  <a href="https://bitbucket.org/cliffmeyers/jenkins-test-public-1.git" class="external-link" target="_blank" rel="nofollow noopener">https://bitbucket.org/cliffmeyers/jenkins-test-public-1.git</a></li>
	<li>Press "Create"</li>
</ol>

<p><b>Expected Results</b></p>
<ul class="alternate" type="square">
	<li>User is taken to Activity tab displaying two builds</li>
</ul>

<p><b>Actual Results</b></p>
<ul class="alternate" type="square">
	<li>User is taken to the tab, but errors in console</li>
</ul>

<p><b>Error</b></p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">Uncaught (in promise) TypeError: Cannot read property <span class="code-quote">'self'</span> of undefined
    at ActivityService.bunkerKey (blueocean.js:5397)
    at ActivityService.setItem (blueocean.js:5440)
    at executeAction (blueocean.js:63849)
    at ActivityService.res (blueocean.js:63821)
    at blueocean.js:5684
</pre>
</div></div>

<p><b>Notes</b>
This error seems to occur because the method receives the ObservableValue instance, instead of the actual object (ObservableValue.value) </p></pre>
</details>
</details>

**Comment by jenkins-infra-bot** [comment]
- _Remote link associated with this issue:_


  - [PR#​1140](https://github.com/jenkinsci/blueocean-plugin/pull/1140)

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/michaelneale.png" title="michaelneale's avatar" /> michaelneale:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-43907?focusedId=298116&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-298116">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>A fix was attempted: </p>

<p><a href="https://github.com/jenkinsci/blueocean-plugin/pull/1027" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/blueocean-plugin/pull/1027</a></p>

<p> </p>

<p>however this broke github events. We need testing coverage before we can do much with this, or <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=cliffmeyers" class="user-hover" rel="cliffmeyers">cliffmeyers</a> to get better. I am lowering priority as this doesn't stop things working when I tried. </p>
</pre>
</details>

</ul>
</details>

<p>A fix was attempted: </p>

<p><a href="https://github.com/jenkinsci/blueocean-plugin/pull/1027" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/blueocean-plugin/pull/1027</a></p>

<p> </p>

<p>however this broke github events. We need testing coverage before we can do much with this, or <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=cliffmeyers" class="user-hover" rel="cliffmeyers">cliffmeyers</a> to get better. I am lowering priority as this doesn't stop things working when I tried. </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jamesdumay.png" title="jamesdumay's avatar" /> jamesdumay:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-43907?focusedId=300487&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-300487">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=cliffmeyers" class="user-hover" rel="cliffmeyers">cliffmeyers</a> low priority but I think you'd be best positioned to take a look after we have some tests in this area written (Ivans work)</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=cliffmeyers" class="user-hover" rel="cliffmeyers">cliffmeyers</a> low priority but I think you'd be best positioned to take a look after we have some tests in this area written (Ivans work)</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/cliffmeyers.png" title="cliffmeyers's avatar" /> cliffmeyers:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-43907?focusedId=302422&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-302422">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I saw the same error at the end of GitHub flow too. My fix is a little less aggressive and GitHub single repo and auto-discover work okay, and this also fixes the error message in Git as originally described in the ticket.</p>
</pre>
</details>

</ul>
</details>

<p>I saw the same error at the end of GitHub flow too. My fix is a little less aggressive and GitHub single repo and auto-discover work okay, and this also fixes the error message in Git as originally described in the ticket.</p>
